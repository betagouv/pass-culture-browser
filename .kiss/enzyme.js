import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      const props = { errors: {} }
      const wrapper = shallow(<MyComponent {...props} />)
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
