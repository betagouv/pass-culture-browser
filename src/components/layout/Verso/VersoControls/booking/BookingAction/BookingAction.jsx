import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'

import Price from '../../../../Price/Price'

class BookingAction extends PureComponent {
  handleBookingAction = () => {
    const { bookingUrl, history } = this.props

    history.push(bookingUrl)
  }

  render() {
    const { isNotBookable, priceRange } = this.props

    return (
      <button
        className="ticket-action"
        disabled={isNotBookable}
        onClick={this.handleBookingAction}
        type="button"
      >
        <Price
          className="ticket-price"
          free="Gratuit"
          value={priceRange}
        />
        <span className="ticket-label">{'Jâ€™y vais !'}</span>
      </button>
    )
  }
}

BookingAction.defaultProps = {
  isNotBookable: false,
}

BookingAction.propTypes = {
  bookingUrl: PropTypes.string.isRequired,
  history: PropTypes.shape().isRequired,
  isNotBookable: PropTypes.bool,
  priceRange: PropTypes.arrayOf(PropTypes.number.isRequired).isRequired,
}

export default BookingAction
