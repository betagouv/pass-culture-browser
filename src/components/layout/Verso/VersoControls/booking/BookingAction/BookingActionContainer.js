import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import { compose } from 'redux'

import BookingAction from './BookingAction'
import getPriceRangeFromStocks from '../../../../../../helpers/getPriceRangeFromStocks'
import selectIsNotBookableByRouterMatch from '../../../../../../selectors/selectIsNotBookableByRouterMatch'
import selectOfferByRouterMatch from '../../../../../../selectors/selectOfferByRouterMatch'
import selectStocksByOfferId from '../../../../../../selectors/selectStocksByOfferId'

export const mapStateToProps = (state, ownProps) => {
  const { match, location } = ownProps
  const { pathname, search } = location

  const bookingUrl = `${pathname}/reservation${search}`
  const isNotBookable = selectIsNotBookableByRouterMatch(state, match)
  const offer = selectOfferByRouterMatch(state, match) || {}
  const stocks = selectStocksByOfferId(state, offer.id)
  const priceRange = getPriceRangeFromStocks(stocks)

  return {
    bookingUrl,
    isNotBookable,
    priceRange,
  }
}

export default compose(
  withRouter,
  connect(mapStateToProps)
)(BookingAction)
