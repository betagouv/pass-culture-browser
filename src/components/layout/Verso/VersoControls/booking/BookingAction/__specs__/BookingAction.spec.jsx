import { mount, shallow } from 'enzyme'
import { createMemoryHistory } from 'history'
import React from 'react'
import { Provider } from 'react-redux'
import { Router } from 'react-router'

import BookingAction from '../BookingAction'
import BookingActionContainer from '../BookingActionContainer'
import getMockStore from '../../../../../../../utils/mockStore'

jest.mock('redux-thunk-data', () => {
  const { requestData } = jest.requireActual('fetch-normalize-data')

  return {
    requestData,
  }
})

describe('src | components | layout | Verso | VersoControls | booking | BookingAction | BookingAction', () => {
  let props

  beforeEach(() => {
    props = {
      history: {
        push: jest.fn(),
      },
      bookingUrl: 'http://booking-layout.com',
      priceRange: [10, 30],
    }
  })

  describe('when the offer is bookable', () => {
    it('should render a price and label within link', () => {
      // given
      props.isNotBookable = false

      // when
      const wrapper = shallow(<BookingAction {...props} />)

      // then
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when the offer is not bookable', () => {
    it('should render a price and label within a wrapper', () => {
      // given
      props.isNotBookable = true

      // when
      const wrapper = shallow(<BookingAction {...props} />)

      // then
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when I click on button for booking', () => {
    it('should render the booking layout', () => {
      // given
      const mockHistory = createMemoryHistory()
      mockHistory.push('/decouverte?param=value')
      const mockStore = getMockStore({
        data: (
          state = {
            bookings: [],
            mediations: [],
            offers: [],
            stocks: [],
          }
        ) => state,
      })
      const wrapper = mount(
        <Provider store={mockStore}>
          <Router history={mockHistory}>
            <BookingActionContainer />
          </Router>
        </Provider>
      )

      // when
      wrapper.find({ children: 'Jâ€™y vais !' }).simulate('click')

      // then
      expect(`${mockHistory.location.pathname}${mockHistory.location.search}`).toBe(
        '/decouverte/reservation?param=value'
      )
    })
  })
})
