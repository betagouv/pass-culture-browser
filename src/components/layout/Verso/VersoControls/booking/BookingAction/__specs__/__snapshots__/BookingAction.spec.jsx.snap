// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | layout | Verso | VersoControls | booking | BookingAction | BookingAction when the offer is bookable should render a price and label within link 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookingAction
    bookingUrl="http://booking-layout.com"
    history={
      Object {
        "push": [MockFunction],
      }
    }
    isNotBookable={false}
    priceRange={
      Array [
        10,
        30,
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Price
          className="ticket-price"
          free="Gratuit"
          value={
            Array [
              10,
              30,
            ]
          }
        />,
        <span
          className="ticket-label"
        >
          J’y vais !
        </span>,
      ],
      "className": "ticket-action",
      "disabled": false,
      "onClick": [Function],
      "type": "button",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "ticket-price",
          "free": "Gratuit",
          "value": Array [
            10,
            30,
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "J’y vais !",
          "className": "ticket-label",
        },
        "ref": null,
        "rendered": "J’y vais !",
        "type": "span",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Price
            className="ticket-price"
            free="Gratuit"
            value={
              Array [
                10,
                30,
              ]
            }
          />,
          <span
            className="ticket-label"
          >
            J’y vais !
          </span>,
        ],
        "className": "ticket-action",
        "disabled": false,
        "onClick": [Function],
        "type": "button",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "ticket-price",
            "free": "Gratuit",
            "value": Array [
              10,
              30,
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "J’y vais !",
            "className": "ticket-label",
          },
          "ref": null,
          "rendered": "J’y vais !",
          "type": "span",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`src | components | layout | Verso | VersoControls | booking | BookingAction | BookingAction when the offer is not bookable should render a price and label within a wrapper 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookingAction
    bookingUrl="http://booking-layout.com"
    history={
      Object {
        "push": [MockFunction],
      }
    }
    isNotBookable={true}
    priceRange={
      Array [
        10,
        30,
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Price
          className="ticket-price"
          free="Gratuit"
          value={
            Array [
              10,
              30,
            ]
          }
        />,
        <span
          className="ticket-label"
        >
          J’y vais !
        </span>,
      ],
      "className": "ticket-action",
      "disabled": true,
      "onClick": [Function],
      "type": "button",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "ticket-price",
          "free": "Gratuit",
          "value": Array [
            10,
            30,
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "J’y vais !",
          "className": "ticket-label",
        },
        "ref": null,
        "rendered": "J’y vais !",
        "type": "span",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Price
            className="ticket-price"
            free="Gratuit"
            value={
              Array [
                10,
                30,
              ]
            }
          />,
          <span
            className="ticket-label"
          >
            J’y vais !
          </span>,
        ],
        "className": "ticket-action",
        "disabled": true,
        "onClick": [Function],
        "type": "button",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "ticket-price",
            "free": "Gratuit",
            "value": Array [
              10,
              30,
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "J’y vais !",
            "className": "ticket-label",
          },
          "ref": null,
          "rendered": "J’y vais !",
          "type": "span",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
